src/ray/raylet/node_manager.cc

```diff
@@ -1869,11 +2022,17 @@ void NodeManager::SubmitTask(const Task &task, const Lineage &uncommitted_lineag
 }
 
 void NodeManager::HandleDirectCallTaskBlocked(const std::shared_ptr<Worker> &worker) {
-  if (!worker || worker->GetAssignedTaskId().IsNil() || worker->IsBlocked()) {
+  if (!worker || /*worker->GetAssignedTaskId().IsNil() ||*/ worker->IsBlocked()) {
     return;  // The worker may have died or is no longer processing the task.
   }
   auto const cpu_resource_ids = worker->ReleaseTaskCpuResources();
   local_available_resources_.Release(cpu_resource_ids);
+  if (USE_NEW_SCHEDULER) {
+    // TODO (ion): replace this hard coded # of CPUs.
+    std::unordered_map<std::string, double> task_request;
+    task_request.emplace("CPU", 1.);
+    new_resource_scheduler_->AddNodeAvailableResources( client_id_.Binary(), task_request);
+  }
```

Comment:
Can we move this block above the !worker check above and return, to cleanly separate the two code paths?

Reply:
done